pkg/cloud/externalconn/connectionpb/connection.proto

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+syntax = "proto3";
+package cockroach.cloud.externalconn.connectionpb;
+option go_package = "connectionpb";
+
+import "gogoproto/gogo.proto";
+import "roachpb/api.proto";
+
+enum ConnectionType {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "TypeUnspecified"];
+  STORAGE = 1 [(gogoproto.enumvalue_customname) = "TypeStorage"];
+}
+
+enum ConnectionProvider {
+  unknown = 0;
+  nodelocal = 1;
+}
+
+message ConnectionDetails {
+  ConnectionProvider provider = 1;
+
+  oneof details {
```

Comment:
I think it makes sense this way because every provider has a different set of params, configurations and information they consider privileged. We will ensure that all these interaction are only through the `ConnectionDetails` interface, but I think having one big proto for all providers would be unwieldy.

Reply:
👍